pkg/route/api/validation/validation.go

```diff
@@ -179,3 +240,71 @@ func validateInsecureEdgeTerminationPolicy(tls *routeapi.TLSConfig, fldPath *fie
 
 	return nil
 }
+
+// validatePrivateKeyForCert checks if a private key PEM is valid and
+// is valid for the certificate (if any).
+func validatePrivateKeyForCert(keyPEM, certPEM string) error {
+	if err := validatePrivateKeyPEM(keyPEM); err != nil {
+		return err
+	}
+
+	if len(certPEM) > 0 {
+		if _, err := tls.X509KeyPair([]byte(certPEM), []byte(keyPEM)); err != nil {
```

Comment:
validating the key+cert is odd inside a `validatePrivateKeyForCert` func... not really sure why we need both `validatePrivateKeyForCert` and `validatePrivateKeyPEM`